src/main/java/com/fmi/cardealership/controller/PaymentController.java

```diff
@@ -0,0 +1,65 @@
+package com.fmi.cardealership.controller;
+
+import com.fmi.cardealership.dto.PaymentDto;
+import com.fmi.cardealership.model.Payment;
+import com.fmi.cardealership.service.PaymentService;
+import org.modelmapper.ModelMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/api/payments")
+public class PaymentController {
+
+    private final PaymentService paymentService;
+    private final ModelMapper modelMapper;
+
+    private static final Logger logger = LoggerFactory.getLogger(PaymentController.class);
+
+    @Autowired
+    public PaymentController(PaymentService paymentService, ModelMapper modelMapper) {
+        this.paymentService = paymentService;
+        this.modelMapper = modelMapper;
+    }
+
+    @GetMapping
+    public List<PaymentDto> getAllPayments() {
+        return paymentService.getAllPayments()
+                .stream()
+                .map(payment -> modelMapper.map(payment, PaymentDto.class))
+                .collect(Collectors.toList());
+    }
+
+    @GetMapping("/{id}")
+    public PaymentDto getPaymentById(@PathVariable Long id) {
+        return modelMapper.map(paymentService.getPaymentById(id), PaymentDto.class);
+    }
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public PaymentDto createPayment(@RequestBody @Valid PaymentDto paymentDto) {
+        Payment payment = modelMapper.map(paymentDto, Payment.class);
+        logger.info(String.format("User with id %d was created",paymentDto.getId()));
+        return modelMapper.map(paymentService.storePayment(payment), PaymentDto.class);
```

Comment:
Here this mapping won`t work. You need to specify how will ids in PaymentDtos be mapped to db models. You can see how to do that at this link: https://www.baeldung.com/java-modelmapper. Look at the section for Converter.